encord/objects/ontology_object_instance.py

```diff
@@ -519,19 +591,34 @@ def get_annotations(self) -> List[Annotation]:
         """
         return [self.get_annotation(frame_num) for frame_num in sorted(self._frames_to_instance_data.keys())]
 
+    def get_annotation_frames(self) -> set[int]:
+        """
+        Get all annotations for the object instance on all frames it has been placed on.
+
+        Returns:
+            List[Annotation]: A list of `ObjectInstance.Annotation` in order of available frames.
+        """
+        if self._range_only:
+            return self._range_manager.get_ranges_as_frames()
+        else:
+            return {self.get_annotation(frame_num).frame for frame_num in sorted(self._frames_to_instance_data.keys())}
+
     def remove_from_frames(self, frames: Frames) -> None:
         """
         Remove the object instance from the specified frames.
 
         Args:
             frames: The frames from which to remove the object instance.
         """
-        frames_list = frames_class_to_frames_list(frames)
-        for frame in frames_list:
-            self._frames_to_instance_data.pop(frame)
+        if self._range_only:
+            raise RuntimeError("NYI")
```

Comment:
Maybe a slightly cleaner exception message that includes contact Encord support

Reply:
fixed, actually forgot to add implementation here